src/pilot/deploy-overcloud.py

```diff
@@ -30,18 +36,534 @@
+                      "ImagePropertiesFilter,ServerGroupAntiAffinityFilter," \
+                      "ServerGroupAffinityFilter,CoreFilter," \
+                      "NUMATopologyFilter," \
+                      "AggregateInstanceExtraSpecsFilter"
+            # Execute the command to check if filters already set
+            ssh_stdout_list, ssh_stderr_list = self.execute_cmd(filter_cmd)
+            if len(ssh_stdout_list) > 0:
+                if filters in ssh_stdout_list[0]:
+                    logger.warning("Nova schedular filters already exists on {} "
+                                   "skipping filters update.".format(self.ip_address))
+                    return
+
+            # Execute the update filter command
+            update_filter = '/usr/bin/crudini --set /etc/nova/nova.conf ' \
+                            'DEFAULT scheduler_default_filters ' + filters
+            ssh_stdout_list, ssh_stderr_list = self.execute_cmd(update_filter)
+            if ssh_stderr_list:
+                raise Exception("Command execution failed "
+                                "with error {}".format(ssh_stderr_list))
+            else:
+                # Execute the restart scheduler service command
+                sched_cmd = "/usr/bin/systemctl " \
+                            "restart openstack-nova-scheduler.service"
+                ssh_stdout_list, ssh_stderr_list = self.execute_cmd(sched_cmd)
+                time.sleep(2)
+
+                if ssh_stderr_list:
+                    logger.error("Failed to restart the scheduler service \
+                    with error".format(ssh_stderr_list))
+        except Exception as error:
+            message = "Exception {}: {}". \
+                      format(type(error).__name__, str(error))
+            logger.error("Failed to update filter on remote machine {} "
+                         "with error {}".format(self.ip_address, message))
+        finally:
+            # Close the ssh connection
+            self.ssh_connection.close()
+
+
+def edit_dell_environment_file(enable_hugepage,
+                               enable_numa,
+                               hugepage_size,
+                               vcpu_pin_set,
+                               dell_compute_count=0):
+    try:
+        logger.info("Editing dell environment file")
+        file_path = home_dir + '/pilot/templates/dell-environment.yaml'
+        if not os.path.isfile(file_path):
+            raise Exception("The dell-environment.yaml file does not exist")
+        cmds = ['sed -i "s|dellnfv::hugepages::enable:.*|' +
+                'dellnfv::hugepages::enable: ' +
+                str(enable_hugepage) +
+                '|" ' +
+                file_path,
+                'sed -i "s|dellnfv::numa::enable:.*|' +
+                'dellnfv::numa::enable: ' +
+                str(enable_numa) +
+                '|" ' +
+                file_path]
+
+        if dell_compute_count > 0:
```

Comment:
I don't think we need the above line since there should always be at least 1 compute

Reply:
Agree. Same comment as above. Removed.